examples/cpp/dds/CustomPayloadPoolExample/CustomPayloadPoolDataSubscriber.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file CustomPayloadPoolDataSubscriber.cpp
+ *
+ */
+
+#include "CustomPayloadPoolDataSubscriber.h"
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/SubscriberAttributes.h>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
+
+using namespace eprosima::fastdds::dds;
+
+CustomPayloadPoolDataSubscriber::CustomPayloadPoolDataSubscriber(
+        std::shared_ptr<CustomPayloadPool> payload_pool)
+    : participant_(nullptr)
+    , subscriber_(nullptr)
+    , topic_(nullptr)
+    , reader_(nullptr)
+    , type_(new CustomPayloadPoolDataPubSubType())
+    , payload_pool_(payload_pool)
+{
+    matched_ = 0;
+    samples_ = 0;
+}
+
+bool CustomPayloadPoolDataSubscriber::init()
+{
+    DomainParticipantQos pqos = PARTICIPANT_QOS_DEFAULT;
+    pqos.name("Participant_sub");
+    auto factory = DomainParticipantFactory::get_instance();
+
+    participant_ = factory->create_participant(0, pqos);
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    //REGISTER THE TYPE
+    type_.register_type(participant_);
+
+    //CREATE THE SUBSCRIBER
+    SubscriberQos sqos = SUBSCRIBER_QOS_DEFAULT;
+
+    subscriber_ = participant_->create_subscriber(sqos, nullptr);
```

Comment:
In general, if we do not modify them, we don't need the local var
```suggestion
    subscriber_ = participant_->create_subscriber(SUBSCRIBER_QOS_DEFAULT, nullptr);
```